homeassistant/components/thethingsnetwork/__init__.py

```diff
@@ -1,29 +1,34 @@
 """Support for The Things network."""
 
+import logging
+
 import voluptuous as vol
 
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_HOST
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import issue_registry as ir
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.typing import ConfigType
 
-CONF_ACCESS_KEY = "access_key"
-CONF_APP_ID = "app_id"
-
-DATA_TTN = "data_thethingsnetwork"
-DOMAIN = "thethingsnetwork"
-
-TTN_ACCESS_KEY = "ttn_access_key"
-TTN_APP_ID = "ttn_app_id"
-TTN_DATA_STORAGE_URL = (
-    "https://{app_id}.data.thethingsnetwork.org/{endpoint}/{device_id}"
+from .const import (
+    CONF_APP_ID,
+    CONF_DEPRECATED_ACCESS_KEY,
+    DOMAIN,
+    PLATFORMS,
+    TTN_API_HOST,
 )
+from .coordinator import TTNCoordinator
+
+_LOGGER = logging.getLogger(__name__)
 
 CONFIG_SCHEMA = vol.Schema(
     {
+        # Configuration via yaml not longer supported - keeping to warn about migration
         DOMAIN: vol.Schema(
             {
                 vol.Required(CONF_APP_ID): cv.string,
-                vol.Required(CONF_ACCESS_KEY): cv.string,
+                vol.Required(CONF_DEPRECATED_ACCESS_KEY): cv.string,
```

Comment:
```suggestion
                vol.Required("access_key"): cv.string,
```
If this is the only place we use it, don't bother